fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanApplicationWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -853,34 +854,21 @@ private Map<String, Object> loanApplicationRejection(Loan loan, final JsonComman
     @Override
     public CommandProcessingResult applicantWithdrawsFromApplication(final Long loanId, final JsonCommand command) {
 
-        final AppUser currentUser = getAppUserIfPresent();
-
-        this.loanApplicationTransitionValidator.validateApplicantWithdrawal(command.json());
-
+        // retrieve loan
         final Loan loan = retrieveLoanBy(loanId);
-        loanApplicationTransitionValidator.checkClientOrGroupActive(loan);
 
+        // validate withdrawal
+        loanApplicationTransitionValidator.validateApplicantWithdrawal(command, loan, defaultLoanLifecycleStateMachine);
```

Comment:
You dont need to send the defaultLoanLifecycleStateMachine as a parameter. You can inject in the loanApplicationTransitionValidator directly